DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/DelegateGroupsControllerTests.cs

```diff
@@ -340,5 +344,289 @@ public void EditGroupName_should_not_update_name_when_linked_to_field_is_not_zer
 
             result.Should().BeNotFoundResult();
         }
+
+        [Test]
+        public void
+            GenerateGroups_should_call_services_and_redirect_after_successfully_generating_groups_from_custom_prompt()

```

Comment:
In line with my comment about the controller, this will all have to shift to service tests

Reply:
Moved the logic tested here to the service tests